### PR TITLE
fix: introduce message strip and depricate alerts

### LINKED ISSUE #2087: Adopt message strip markup update
#### Is this a bug, enhancement, or feature request?
enhancement

#### Briefly describe your proposal.
https://github.com/SAP/fundamental-styles/pull/714
dismiss button comes after the message

### PR DESCRIPTION
**Pull request by InnaAtanasova**
#### Please provide a link to the associated issue.
closes #2087 

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
This is the same PR as https://github.com/SAP/fundamental-ngx/pull/2153 with changes from the latest master.

**Comment by stefanoScalzo** [comment]
I know that it is shown in the code but can you write in the docs what the different types they can choose from please

**Comment by stefanoScalzo** [comment]
Also, what happened to the 'alert' / message strip with timeouts?

**Comment by InnaAtanasova** [comment]
> Also, what happened to the 'alert' / message strip with timeouts?

Message Strips are inline messages, they won't show with overlay. Message Toast, Message Box, Dialog are used instead

**Comment by InnaAtanasova** [comment]
> I know that it is shown in the code but can you write in the docs what the different types they can choose from please

We have API docs for this purpose. But anyways, I had to add some other stuff so I added it to the docs

File: apps/docs/src/app/core/component-docs/alert/alert-header/alert-header.component.html [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <header>Alert</header>
+<fd-alert [type]="'warning'">
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/alert/alert-header/alert-header.component.html:2]: Should we use message strip for these warnings?

File: apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+</fd-docs-section-title>
+<description>A number of message strip types are available. They offer preset styling for different contextual situations. Can be one of 'warning', 'success', 'information', 'error' or normal (default). </description>
+<component-example>
+    <fd-message-strip-example></fd-message-strip-example>
+</component-example>
+<code-example [exampleFiles]="messageStripBasicExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title [id]="'message-strip-types'" [componentName]="'messageStrip'">
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html:12]: Repeating `[id]` value, also for "Message Strips With Customizable Width"

File: apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+<description>The default message strip has no icon. Information, Success, Warning and Error message strips by default are rendered with icons. To remove them set the <code>[noIcon]</code> property to <code>true</code>. </description>
+<component-example>
+    <fd-message-strip-noicon-example></fd-message-strip-noicon-example>
+</component-example>
+<code-example [exampleFiles]="messageStripNoIconExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title [id]="'message-strip-types'" [componentName]="'messageStrip'">
+    Message Strips With Customizable Width
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html:24]: @Vanessa-Cusmich Should it be:
"Message Strips With Customizable Width" => "Message Strips With Customized Width"

File: libs/core/src/lib/message-strip/message-strip.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        fd-button
+        [compact]="true"
+        [fdType]="'transparent'"
+        *ngIf="dismissible"
+        (click)="dismiss()"
+        [attr.aria-controls]="id"
+        [attr.aria-label]="dismissLabel">
+</button>
+<p class="fd-message-strip__text">
+    <ng-container #container>{{message}}</ng-container>
```
Comment (salarenko) [inline libs/core/src/lib/message-strip/message-strip.component.html:11]: What is the purpose of this `<ng-container #container></ng-container>`

File: libs/core/src/lib/message-strip/message-strip.component.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        '[style.min-width]': 'minWidth',
+        'role': 'alert',
+        '[attr.id]': 'id',
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MessageStripComponent implements OnInit, OnChanges, CssClassBuilder {
+
+    /** user's custom classes */
```
Comment (salarenko) [inline libs/core/src/lib/message-strip/message-strip.component.ts:26]: Start from big letter

File: libs/core/src/lib/message-strip/message-strip.component.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    @Input()
+    minWidth: string;
+
+    /** Alternative way of passing in a message to the message-strip. */
+    @Input()
+    message: string;
+
+    /** Event fired when the message-strip is dismissed. */
+    @Output()
+    onDismiss: EventEmitter<undefined> = new EventEmitter<undefined>();
```
Comment (salarenko) [inline libs/core/src/lib/message-strip/message-strip.component.ts:77]: Its barely no difference, but I think we should follow current convention and go with:
```
    onDismiss: EventEmitter<void> = new EventEmitter<void>();
```

File: libs/core/src/lib/message-strip/message-strip.component.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        this.buildComponentCssClass();
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
```
Comment (salarenko) [inline libs/core/src/lib/message-strip/message-strip.component.ts:93]: Group life cycle hooks methods after constructor

File: libs/core/src/lib/message-strip/message-strip.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    it('Should Add no-icon modifier class', () => {
+        component.ngOnInit();
+        component.noIcon = true;
+        component.ngOnChanges();
+        fixture.detectChanges();
+        expect(component.elementRef().nativeElement.classList.contains('fd-message-strip--no-icon')).toBe(true);
+    });
+
+    it('Should apply a type', () => {
```
Comment (salarenko) [inline libs/core/src/lib/message-strip/message-strip.component.spec.ts:46]: Missing test for dismissing message strip

**Comment by InnaAtanasova** [comment]
> I don't know how, but somehow button label is `title`.
> Should be `dismissLabel: string = 'Dismiss'`.
> 
> image
> 

This issue is coming from Button component @rengare 
Screen Shot 2020-04-06 at 12 36 02 PM

File: apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html (hunk 3 above)
Comment (Vanessa-Cusmich) [reply apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html:24]: @salarenko it should be customizable.

File: apps/docs/src/app/core/component-docs/alert/alert-header/alert-header.component.html (hunk 1 above)
Comment (InnaAtanasova) [reply apps/docs/src/app/core/component-docs/alert/alert-header/alert-header.component.html]: Done

File: apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html (hunk 2 above)
Comment (InnaAtanasova) [reply apps/docs/src/app/core/component-docs/message-strip/message-strip-docs.component.html]: Fixed

File: libs/core/src/lib/message-strip/message-strip.component.html (hunk 4 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-strip/message-strip.component.html]: Removed

File: libs/core/src/lib/message-strip/message-strip.component.ts (hunk 5 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-strip/message-strip.component.ts]: Fixed

File: libs/core/src/lib/message-strip/message-strip.component.ts (hunk 6 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-strip/message-strip.component.ts]: Fixed

File: libs/core/src/lib/message-strip/message-strip.component.ts (hunk 7 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-strip/message-strip.component.ts]: Done

**Comment by InnaAtanasova** [comment]
> I also have doubts this is correct to to only visually hide message-strip after dismissing it. Shouldn't we rather remove it from the DOM?
I didn't introduce a new way, I followed the implementation from Alert. If you suggest a new way we can discuss it during the daily meeting.

File: libs/core/src/lib/message-strip/message-strip.component.spec.ts (hunk 8 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-strip/message-strip.component.spec.ts:46]: Added

**Pull request merged**